### PR TITLE
chore: #629 migrate config of apps to config.json

### PR DESCRIPTION
**Pull request by tanphamhaiduong**
Changes:
- Add config.example.json
- Add method to fetch config.json from public folder for webpack
- Change `process.env. ...` to `window.reapit.config. ...`

### CODE REVIEW

File: packages/marketplace/src/types/global.d.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+export type Config = {
+  appEnv: string
```
Comment (willmcvay) [inline packages/marketplace/src/types/global.d.ts]: Should be a type like 'local' | 'dev' | 'prod' etc right?

File: packages/marketplace/src/components/hocs/error-boundary.tsx [hunk 2]
```diff
@@ -40,7 +40,8 @@ export class ErrorBoundary extends React.Component<ErrorProps, ErrorState> {
       type: 'COMPONENT',
       message: errorMessages.DEFAULT_COMPONENT_ERROR,
     })
-    if (process.env.NODE_ENV === 'production') {
+    const isDevelopment = window.reapit.config.appEnv === 'development'
```
Comment (willmcvay) [inline packages/marketplace/src/components/hocs/error-boundary.tsx]: So what is the app env for local development? I mean, sometimes we need to differentiate between local and development?

File: packages/marketplace/src/components/pages/login.tsx [hunk 3]
```diff
@@ -48,8 +48,7 @@ export const tabConfigs = ({ loginType, history }: LoginProps): TabConfig[] => [
 ]
 
 export const Login: React.FunctionComponent<LoginProps> = (props: LoginProps) => {
-  const reapitEnv = process.env.REAPIT_ENV || 'LOCAL'
-  const isReapitEnvProd = reapitEnv === 'PROD'
+  const isProduction = window.reapit.config.appEnv === 'production'
```
Comment (willmcvay) [inline packages/marketplace/src/components/pages/login.tsx]: In this case we need to differentiate between dev and prod - only prod hides the login options tabs

File: packages/marketplace/src/core/router.tsx [hunk 4]
```diff
@@ -32,9 +32,9 @@ const RegisterConfirm = React.lazy(() => import('../components/pages/register-co
 const AdminStats = React.lazy(() => import('../components/pages/admin-stats'))
 
 const Router = () => {
-  const isReapitEnvProd = process.env.REAPIT_ENV === 'PROD'
+  const isDevelopment = window.reapit.config.appEnv === 'development'
```
Comment (willmcvay) [inline packages/marketplace/src/core/router.tsx]: What about local dev?

Comment (tanphamhaiduong) [reply packages/marketplace/src/core/router.tsx]: I think local dev env is the same with development environment so I let 2 things same

File: packages/marketplace/src/components/pages/login.tsx (hunk 3 above)
Comment (tanphamhaiduong) [reply packages/marketplace/src/components/pages/login.tsx]: yes only production hide login option tabs

**Pull request merged**